src/stratigraphy/main.py

```diff
@@ -37,474 +25,67 @@
-
-                with fitz.Document(in_path) as doc:
-                    for page_index, page in enumerate(doc):
-                        page_number = page_index + 1
-                        logger.info("Processing page %s", page_number)
-
-                        predictions, depths_materials_column_pairs = process_page(page, **params)
-
-                        output[filename][f"page_{page_number}"] = {
-                            "layers": predictions,
-                            "depths_materials_column_pairs": depths_materials_column_pairs,
-                        }
-        return output
-
-
-if __name__ == "__main__":
-    # setup mlflow tracking; should be started before any other code
-    # such that tracking is enabled in other parts of the code.
+@click.command()
+@click.option(
+    "-i",
+    "--input_directory",
+    type=click.Path(exists=True, path_type=Path),
+    default=DATAPATH / "Benchmark",
+    help="Path to the input directory.",
+)
+@click.option(
+    "-g",
+    "--ground_truth_path",
+    type=click.Path(exists=True, path_type=Path),
+    default=DATAPATH / "Benchmark" / "ground_truth.json",
+    help="Path to the ground truth file.",
+)
+@click.option(
+    "-o",
+    "--out_directory",
+    type=click.Path(path_type=Path),
+    default=DATAPATH / "Benchmark" / "evaluation",
+    help="Path to the output directory.",
+)
+@click.option(
+    "-p",
+    "--predictions_path",
+    type=click.Path(path_type=Path),
+    default=DATAPATH / "Benchmark" / "extract" / "predictions.json",
+    help="Path to the predictions file.",
+)
+@click.option(
+    "-s",
+    "--skip-draw-predictions",
+    is_flag=True,
+    default=False,
+    help="Whether to skip drawing the predictions on pdf pages. Defaults to False.",
+)
+@click.option(
+    "-l", "--draw-lines", is_flag=True, default=False, help="Whether to draw lines on pdf pages. Defaults to False."
+)
+def start_pipeline(
+    input_directory: Path,
+    ground_truth_path: Path,
+    out_directory: Path,
+    predictions_path: Path,
+    skip_draw_predictions: bool = False,
+    draw_lines: bool = False,
+):
+    """Run the boreholes data extraction pipeline.
+
+    The pipeline will extract material description of all found layers and assign them to the corresponding
+    depth intervals. The input directory should contain pdf files with boreholes data. The algorithm can deal
+    with borehole profiles of multiple pages.
+
+    Args:\n
+        input_directory (Path): The directory containing the pdf files.\n
+        ground_truth_path (Path): The path to the ground truth file json file.\n
+        out_directory (Path): The directory to store the evaluation results.\n
+        predictions_path (Path): The path to the predictions file.\n
+        skip_draw_predictions (bool, optional): Whether to skip drawing predictions on pdf pages. Defaults to False.\n
+        draw_lines (bool, optional): Whether to draw lines on pdf pages. Defaults to False.\n
```

Comment:
Isn't there a way to generate the docstrings correctly without these explicit "\n"?

Reply:
The problem is related to how click provides the docstring to stdout when prompting `boreholes-extract-materials --help`. Without the `\n` it's all on one line which is not very readable.

Reply:
But let me dig a bit.